### PR TITLE
Use proper case for year date format

### LINKED ISSUE #40493: Date Filtering Bug in v0.49.1 Docker Image with Japanese Locale Settings
### Describe the bug

A bug occurs in both the v0.49.0 and v0.49.1 Docker images uploaded to DockerHub when the language setting is switched to Japanese. Specifically, when filtering datetime columns by specifying an exact date, the filter incorrectly applies the date subtracted by 2018 years from the chosen date. This issue has been confirmed not to occur in version v0.48.8. Additionally, the bug does not manifest when the language settings are set to English, Chinese, or Korean.

When a date is selected like this:
image

The filter incorrectly applies a date subtracted by 2018 years as shown below:
image

### To Reproduce

1. Use the v0.49.1 image from DockerHub.
2. Change language settings to Japanese.
3. Attempt to filter a datetime column by specifying an exact date.
4. Notice that the applied filter date is 2018 years less than the specified date.

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "ja-JP",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Linux",
    "os.version": "5.10.210-201.852.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "8.0.35"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.7.10"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-03-21",
      "tag": "v0.49.1",
      "hash": "54ef5e9"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

P2 - High: Critically blocks data analysis for all Japanese locale users by improperly applying date filters.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nemanjaglumac**
Fixes #40493

### Description

This PR fixes a regression caused by improper year token formatting in momentJS. It is a bit silly, but if we look at [the documentation](https://momentjs.com/docs/#/parsing/string-format/), it shows that year must be represented with all caps `YYYY` token. It also says that "Tokens are case-sensitive." 

### How to verify

Follow the repro steps from the original issue.

### Demo
https://github.com/metabase/metabase/assets/31325167/8ef769d9-03d6-46f6-b0d4-e004d69b9597

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

Running `filter.unit.spec.ts`

1. without a fix
```
Test Suites: 1 failed, 1 total
Tests:       30 failed, 296 passed, 326 total
Snapshots:   0 total
Time:        6.412 s
```

2. with the fix
```
Test Suites: 1 passed, 1 total
Tests:       326 passed, 326 total
Snapshots:   0 total
Time:        4.882 s, estimated 5 s
```

### CODE REVIEW

**Review by romeovs — APPROVED** [review]

File: frontend/src/metabase-lib/filter.unit.spec.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import moment from "moment-timezone"; // eslint-disable-line no-restricted-imports -- deprecated usage
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/filter.unit.spec.ts:1]: we shouldn't be using `moment-timezone` as it's deprecated

Comment (nemanjaglumac) [reply frontend/src/metabase-lib/filter.unit.spec.ts:1]: We switched to dayjs, but migrating the **existing** usages of moment to it should be a task in itself.

Comment (uladzimirdev) [reply frontend/src/metabase-lib/filter.unit.spec.ts:1]: true, but we should try not to introduce new usages of deprecated moment-timezone if possible

Comment (nemanjaglumac) [reply frontend/src/metabase-lib/filter.unit.spec.ts:1]: How would you test something that directly depends on the `moment`?

Comment (uladzimirdev) [reply frontend/src/metabase-lib/filter.unit.spec.ts:1]: we probably need to add more specific assert

Comment (uladzimirdev) [reply frontend/src/metabase-lib/filter.unit.spec.ts:1]: @nemanjaglumac actually I missed that we use moment for formatting under the hood

```
function serializeDate(date: Date): string {
  return moment(date).format(DATE_FORMAT);
}
```

so my apologies

Comment (nemanjaglumac) [reply frontend/src/metabase-lib/filter.unit.spec.ts:1]: No worries.

**Pull request merged**